discovery-provider/src/queries/get_health.py

```diff
@@ -251,3 +249,22 @@ def get_latest_chain_block_set_if_nx(redis=None, web3=None):
             logger.error(f"Could not set values in redis for get_latest_chain_block_set_if_nx: {e}")
 
     return latest_block_num, latest_block_hash
+
+# returns a tuple (<boolean>, <boolean>) that represents whether
+# the given blockhash is present, the given blocknumber is passed
+def get_block_confirmation(blockhash, blocknumber):
+    db = db_session.get_db_read_replica()
+    with db.scoped_session() as session:
+        blockhash_query = session.query(Block).filter(
+            Block.blockhash == blockhash
+        ).all()
+
+        latest_block_query = session.query(Block).filter(
+            Block.is_current == True
+        ).all()
+        assert len(latest_block_query) == 1, "Expected SINGLE row marked as current"
```

Comment:
this is a good check, but might be worth try catching on the route side and returning an error response if this throws rather than the default which i think would just be 500 internal server error

Reply:
cool cool